src/Core/Domain/CmsPageCategory/ValueObject/CmsPageCategoryId.php

```diff
@@ -0,0 +1,63 @@
+<?php
+/**
+ * 2007-2018 PrestaShop.
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Domain\CmsPageCategory\ValueObject;
+
+use PrestaShop\PrestaShop\Core\Domain\CmsPageCategory\Exception\CmsPageCategoryException;
+
+/**
+ * Class CmsPageCategoryId is responsible for providing identificator for cms page category
+ */
+class CmsPageCategoryId
+{
+    /**
+     * @var int
+     */
+    private $cmsPageCategoryId;
+
+    /**
+     * @param int $cmsPageCategoryId
+     *
+     * @throws CmsPageCategoryException
+     */
+    public function __construct($cmsPageCategoryId)
+    {
+        if (!is_numeric($cmsPageCategoryId) || 0 >= $cmsPageCategoryId) {
+            throw new CmsPageCategoryException(
+                sprintf('Invalid cms page category id %s supplied', var_export($cmsPageCategoryId, true))
+            );
+        }
+        $this->cmsPageCategoryId = (int) $cmsPageCategoryId;
```

Comment:
I'm wondering ... maybe we should just refuse not-integer values for `$cmsPageCategoryId` ? Forcing people to provide a real integer will make their code cleaner. Accepting values such as `"42"` looks kind but expose them to later errors

Reply:
I can agree with you - just there will be more type casting when the id will be passed for now since the ObjectModel instead of giving integer it gives string if I recall correctly

Reply:
If ObjectModel provides bad input, it should be treated as bad and processed accordingly 😄